docs/content/utilities/sqs_large_message_handling.mdx

```diff
@@ -0,0 +1,106 @@
+---
+title: SQS Large Message Handling
+description: Utility
+---
+
+The large message handling utility handles SQS messages which have had their payloads
+offloaded to S3 due to them being larger than the SQS maximum.
+
+The utility automatically retrieves messages which have been offloaded to S3 using the
+[amazon-sqs-java-extended-client-lib](https://github.com/awslabs/amazon-sqs-java-extended-client-lib)
+client library. Once the message payloads have been processed successful the
+utility can delete the message payloads from S3.
+
+This utility is compatible with versions *1.1.0+* of amazon-sqs-java-extended-client-lib.</p>
+
+```xml
+<dependency>
+    <groupId>com.amazonaws</groupId>
+    <artifactId>amazon-sqs-java-extended-client-lib</artifactId>
+    <version>1.1.0</version>
+</dependency>
+```
+
+## Install
+
+To install this utility, add the following dependency to your project.
+
+```xml
+<dependency>
+    <groupId>software.amazon.lambda</groupId>
+    <artifactId>powertools-sqs</artifactId>
+    <version>x.x.x</version>
+</dependency>
+```
+
+And configure the aspectj-maven-plugin to compile-time weave (CTW) the
+aws-lambda-powertools-java aspects into your project. You may already have this
+plugin in your pom. In that case add the depenedency to the `aspectLibraries`
+section.
+
+```xml
+<build>
+    <plugins>
+        ...
+        <plugin>
+             <groupId>org.codehaus.mojo</groupId>
+             <artifactId>aspectj-maven-plugin</artifactId>
+             <version>1.11</version>
+             <configuration>
+                 <source>1.8</source>
+                 <target>1.8</target>
+                 <complianceLevel>1.8</complianceLevel>
+                 <aspectLibraries>
+                     ...
+                     <aspectLibrary>
+                         <groupId>software.amazon.lambda</groupId>
+                         <artifactId>powertools-sqs</artifactId>
+                     </aspectLibrary>
+                     ...
+                 </aspectLibraries>
+             </configuration>
+             <executions>
+                 <execution>
+                     <goals>
+                         <goal>compile</goal>
+                     </goals>
+                 </execution>
+             </executions>
+        </plugin>
+        ...
+    </plugins>
+</build>
+```
+
+The annotation `@LargeMessageHandler` should be used with the handleRequest method of a class
+which implements `com.amazonaws.services.lambda.runtime.RequestHandler` with
+`com.amazonaws.services.lambda.runtime.events.SQSEvent` as the first parameter.
+
+```java
+public class SqsMessageHandler implements RequestHandler<SQSEvent, String> {
+
+    @Override
+    @LargeMessageHandler
+    public String handleRequest(SQSEvent sqsEvent, Context context) {
+    // process messages
+
+    return "ok";
+}
+```
+
+`@LargeMessageHandler` creates a default S3 Client `AmazonS3 amazonS3 = AmazonS3ClientBuilder.defaultClient()`.
```

Comment:
May be mention a bit explicitly that if download of any message fails from S3 while enriching, entire batch will fail?